care/facility/api/serializers/patient_consultation.py

```diff
@@ -165,8 +184,46 @@ def update(self, instance, validated_data):
 
         consultation = super().update(instance, validated_data)
 
+        try:
+            health_details_data = self.context["request"].data[
+                "new_health_details"
+            ]
+            vaccination_history = health_details_data.pop(
+                "vaccination_history", []
+            )
+            PatientHealthDetails.objects.filter(
+                id=consultation.last_health_details.id
+            ).update(**self.context["request"].data["new_health_details"])
```

Comment:
use validated data, do not use raw data from the request

Reply:
@vigneshhari I am using the value from `new_health_details` (not a field in consultation API) to create a `health_detail` record. The `validated_data` will remove the `new_health_details` field that's why I am passing the data this way.